content/300-guides/300-upgrade-guides/200-upgrading-versions/050-upgrading-to-prisma-3/index.mdx

```diff
@@ -0,0 +1,78 @@
+metaTitle: 'Upgrade to Prisma 3'
+metaDescription: 'Guides on how to upgrade to Prisma 3'
+tocDepth: 2
+toc: true
+---
+
+<TopBlock>
+
+Prisma 3 introduces a number of **breaking changes** if you are upgrading from an earlier version (any version pre 3.x), therefore, it is important to understand how this upgrade might affect your application.
+
+Below you will find a list of the breaking changes and how to handle them.
+
+</TopBlock>
+
+## Breaking changes
+
+There are a number of breaking changes to be aware of when upgrading from version 2.x to 3.x.
+
+### [Referential actions](../../../../concepts/components/prisma-schema/relations/referential-actions)
+
+The introduction of referential actions in version 3.x removes the safety net in the Prisma Client that had previously prevented cascading deletes at runtime.
+
+As a result, depending on which workflow you are using to work on your application, you could be impacted. We advise you to check your schema and decide if you need to define referential actions explicitly.
+
+See [Referential action upgrade paths](./referential-actions) to understand how to proceed.
+
+### [$queryRaw](../../../../concepts/components/prisma-client/raw-database-access)
+
+From version 3.x onwards, the `$queryRaw` method now only supports a template literal.
+
+This means that if your application relied on `$queryRaw` calls using _strings_, those calls will **not** work anymore. We advise you to use template literals wherever possible for security reasons or resort to `$queryRawUnsafe` otherwise, after carefully escaping queries to prevent SQL injections.
+
+You can learn more about the new `$queryRaw` and `$queryRawUnsafe` methods in the [Raw database access](../../../../concepts/components/prisma-client/raw-database-access) section of the docs.
+
+### [Named constraints](../../../../concepts/components/prisma-schema/names-in-underlying-database)
+
+We changed the convention followed by Prisma to name constraints and indexes. We also introduced a `map` attribute in the PSL to allow defining these names explicitly from the Prisma schema.
+This means that you will notice an impact when running Prisma `migrate` or `db pull` which will follow this new convention. We advise you to adjust your schema to reflect the names of your constraints and indexes appropriately.
+
+You can check out the [Named constraints upgrade path](./named-constraints) for more information on how to proceed.
+
+## Specific upgrade paths
+
+<Subsections />
+
+## Upgrading the <inlinecode>prisma</inlinecode> and <inlinecode>@prisma/client</inlinecode> packages to Prisma 3
+
+To upgrade from version 2.x to 3.x, you need to update both the `prisma` and `@prisma/client` packages. Both the `prisma` and `@prisma/client` packages install with a caret `^` in their version number to safe guard against breaking changes.
+
+To ignore the caret `^` and upgrade across major versions, you can use the `@3` tag when upgrading with `npm`, or `yarn` .
```

Comment:
@janpio did mention that `@latest` was needed in a recent message. Shouldn't that be aligned properly?

Reply:
`@3` also works, but in the Update Notifier we go with `@latest` as that is the goal - getting to the most recent version. 

In the context of a Prisma 3 upgrade guide, it can also be fine to use `@3` as that will also do exactly what is descried in the guide when we release Prisma 4.x one day. On the other hand, having a unified set of commands we teach people also has benefits.

Reply:
Yep, if we have `@latest` and someone comes here from Prisma 2.x, they could accidently install 4.x with `@latest`. Of course, you really want to be up to date, but this guide is specific to 2.x - 3.x including the relevant breaking changes so it made sense to update them within that major